wagtail/admin/templates/wagtailadmin/workflows/task_chooser/chooser.html

```diff
@@ -30,7 +30,16 @@
                 aria-labelledby="tab-label-existing"
                 hidden
             >
-                <form class="task-search" action="{% url 'wagtailadmin_workflows:task_chooser_results' %}" method="GET" novalidate>
+                <form
+                    class="task-search"
+                    action="{% url 'wagtailadmin_workflows:task_chooser_results' %}"
+                    method="GET"
+                    novalidate
+                    data-controller="w-swap"
+                    data-action="submit->w-swap#submit:stop:prevent change->w-swap#submitLazy input->w-swap#submitLazy navigate->w-swap#replace"
```

Comment:
There's one small issue that I noticed with this:

When there's pagination, if you type in the search field and (without changing focus away from the input) you click on the "next" link, you'll need to click it twice. This is because the `change` event is fired.

In the old implementation, the `input` event listener is only attached to the `input` elements, and the `change` event listener is only attached to the "filter" elements, so this issue did not exist.

Not quite sure what's the best way to proceed with this, as we're attaching the listeners on the form rather than individual elements. Probably minor enough to be ignored, but would be nice if there's a way to fix this.

FYI the "filter" is possible here when there's multiple task types (https://docs.wagtail.org/en/stable/extending/custom_tasks.html)

and to easily test the pagination, I changed the pagination to have 2 per page:
https://github.com/wagtail/wagtail/blob/0f97838381d7c6d5f30d0b9b2f39447a559d10d2/wagtail/admin/views/workflows.py#L732

One workaround would be to check if the serialised form is identical to the one that was last submitted, but probably don't check it if it's the `submit` method so user can force a "refresh" by hitting enter. This is probably overengineering, though.

Reply:
Nice find, I think I have fixed this.

Made two changes; moved the navigate to before the others, while this technically should not matter it just means it will start triggering before any lazy handlers.

The main fix is just removing the `change` event listener as this is overkill for what we are doing here, probably legacy from the first IE11 implementations where `input` sometimes had rough edges. Basically `input` will trigger on any value change as the user types (or selects something in a select field), but `change` will fire once an `input` field looses focus.

The same problem could technically happen but the user would need to click 'next' within the 50ms window or before the partial async response has come back. Although it's possible that could have happened with the previous implementation.

https://javascript.info/events-change-input#event-input